File: src/main/java/com/musinsa/subject/service/facade/BrandFacade.java
package com.musinsa.subject.service.facade;

import com.musinsa.subject.model.brand.BrandRequest;
import com.musinsa.subject.model.brand.BrandResponse;

public interface BrandFacade {

    BrandResponse createBrand(BrandRequest request);

    BrandResponse getBrand(long brandId);

    BrandResponse updateBrand(long brandId, BrandRequest request);

    void deleteBrand(long brandId);

}

File: src/main/kotlin/com/musinsa/subject/service/facade/BrandFacade.kt
package com.musinsa.subject.service.facade

import com.musinsa.subject.model.brand.BrandRequest
import com.musinsa.subject.model.brand.BrandResponse

interface BrandFacade {
    fun createBrand(request: BrandRequest): BrandResponse
    fun getBrand(brandId: Long): BrandResponse
    fun updateBrand(brandId: Long, request: BrandRequest): BrandResponse
    fun deleteBrand(brandId: Long)
}
